elyra/pipeline/parser.py

```diff
@@ -30,37 +31,38 @@ def __init__(self, root_dir="", **kwargs):
         super().__init__(**kwargs)
         self.root_dir = root_dir
 
-    def parse(self, pipeline_definitions: Dict) -> Pipeline:
+    def parse(self, pipeline_json: Dict) -> Pipeline:
         """
         The pipeline definitions allow for defining multiple pipelines  in one json file.
         When super_nodes are used, their node actually references another pipeline in the
         set of pipeline definitions - which is "flattened" into the overall pipeline object's
         list of operations.
         """
 
-        primary_pipeline_id = pipeline_definitions['primary_pipeline']
-        primary_pipeline = PipelineParser._get_pipeline_definition(pipeline_definitions, primary_pipeline_id)
+        pipeline_definition = PipelineDefinition(pipeline_definition=pipeline_json)
+
+        primary_pipeline = pipeline_definition.primary_pipeline
 
         # runtime info is only present on primary pipeline...
-        runtime = PipelineParser._get_app_data_field(primary_pipeline, 'runtime')
+        runtime = primary_pipeline.runtime
         if not runtime:
             raise ValueError("Invalid pipeline: Missing runtime.")
-        runtime_config = PipelineParser._get_app_data_field(primary_pipeline, 'runtime-config')
+        runtime_config = primary_pipeline.runtime_config
         if not runtime_config:
             raise ValueError("Invalid pipeline: Missing runtime configuration.")
 
-        source = PipelineParser._get_app_data_field(primary_pipeline, 'source')
+        source = primary_pipeline.source
 
-        pipeline_object = Pipeline(id=primary_pipeline_id,
-                                   name=PipelineParser._get_app_data_field(primary_pipeline, 'name', 'untitled'),
+        pipeline_object = Pipeline(id=primary_pipeline.id,
+                                   name=primary_pipeline.name,
                                    runtime=runtime,
                                    runtime_config=runtime_config,
                                    source=source)
-        self._nodes_to_operations(pipeline_definitions, pipeline_object, primary_pipeline['nodes'])
+        self._nodes_to_operations(pipeline_json, pipeline_object, primary_pipeline.to_dict().get('nodes'))
         return pipeline_object
 
     def _nodes_to_operations(self,
-                             pipeline_definitions: Dict,
+                             pipeline_definition: Dict,
```

Comment:
Shouldn't this use the helper classes?
```suggestion
                             pipeline_definition: PipelineDefintion,
```